latte/models/cca/estimation.py

```diff
@@ -56,7 +56,9 @@ def fit(X: np.ndarray, Z: np.ndarray, subspace_size: int, p_train: float = 0.8)
     cca.fit(X_train, Z_train)
 
     A_hat_1, A_hat_2 = cca.x_weights_, cca.y_weights_
-    E_hat_1, E_hat_2 = np.eye(A_hat_1.shape[0]) - A_hat_1 @ A_hat_1.T, np.eye(A_hat_2.shape[0]) - A_hat_2 @ A_hat_2.T
+    (d_1, m_1), (d_2, m_2) = A_hat_1.shape, A_hat_2.shape
+    E_hat_1 = np.linalg.svd(np.eye(A_hat_1.shape[0]) - A_hat_1 @ A_hat_1.T, full_matrices=False)[0][:, : d_1 - m_1]
+    E_hat_2 = np.linalg.svd(np.eye(A_hat_2.shape[0]) - A_hat_2 @ A_hat_2.T, full_matrices=False)[0][:, : d_2 - m_2]
```

Comment:
(As above, `np.eye(d_2)` may read better).

Reply:
Actually, I've noticed that the pattern
`svd(eye(x) - y)[0][:, :, z]`
appears several times – perhaps making it into a subroutine (with a comment on `[0][:, :, z]`) would be helpful :slightly_smiling_face: